### DISCUSSION TITLE
Passing data between child nodes

### DISCUSSION THREAD

**Discussion by Wysel** [bumble-tech/appyx#148] [General]
Is there a recommended way of passing data between nodes?

**Comment by KovalevAndrey** [comment]
Hi @Wysel, we have an ChildAware API which helps you understand when any particular child has been attached to the ParentNode. Then it's up to you how you set up communication between parent and child. You can use simple callbacks or use [reactive API](https://github.com/bumble-tech/appyx/blob/main/interop-rx2/src/main/kotlin/com/bumble/appyx/interop/rx2/connectable/NodeConnector.kt) for that. You can check an example how we do it using [MVICore and binder](https://github.com/bumble-tech/appyx/blob/main/sandbox/src/main/kotlin/com/bumble/appyx/sandbox/client/mvicoreexample/MviCoreExampleInteractor.kt).

Note that because Appyx app is effectively a tree, therefore horizontal communication between children is discouraged, and should happen via the `ParentNode`.

**Reply by Wysel** [reply]
Hi @KovalevAndrey, I'm building child communication through a ParentNode, just trying to avoid reinventing a wheel just in case that's already implemented within the library.

BTW, when do you expect to have api documentation available?

Thanks.

**Reply by KovalevAndrey** [reply]
@Wysel it's coming soon, probably next week
